src/compiler/checker.ts

```diff
@@ -1902,11 +1902,16 @@ namespace ts {
                 resolveEntityName(node.propertyName || node.name, meaning, /*ignoreErrors*/ false, dontResolveAlias);
         }
 
-        function getTargetOfExportAssignment(node: ExportAssignment, dontResolveAlias: boolean): Symbol {
-            return resolveEntityName(<EntityNameExpression>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace, /*ignoreErrors*/ false, dontResolveAlias);
+        function getTargetOfExportAssignment(node: ExportAssignment, dontResolveAlias: boolean): Symbol | undefined {
+            const aliasLike = resolveEntityName(<EntityNameExpression>node.expression, SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace, /*ignoreErrors*/ true, dontResolveAlias);

```

Comment:
Seems like passing this would not pass nameNotFoundMessage to resolveName and in turn the resolved symbol wont have result.isReferenced set...

Reply:
@sheetalkamat I just pushed a change removing the check for `nameNotFoundMessage` for setting `symbol.isReferenced` from `resolveName` - there's an `isUse` parameter being checked already, so it didn't even seem necessary to check; and the tests seem to bear that out (and we do have places where we explicitly say there's no error message, but `isUse` is true - this is the only thing `isUse` controls - so it seems like it is intended for it to be set independently of the error message).